### ISSUE TITLE
Handle empty publish maps more gracefully

### ISSUE THREAD

**Issue by michaelklishin** [rabbitmq/rabbitmq-shovel#46]
Currently a Shovel declared with an empty publish property map fails to initialise:

```
2018-11-26 08:16:47.710 [error] <0.5043.0> Supervisor {<0.5043.0>,rabbit_shovel_dyn_worker_sup} had child {<<"/">>,<<"shovel1">>} started with
rabbit_shovel_worker:start_link(dynamic, {<<"/">>,<<"shovel1">>}, [{<<"add-forward-headers">>,true},{<<"dest-exchange">>,<<"exchange1">>},{<<"dest-uri">>,<<"amqp://...">>},...])
at undefined exit with reason no function clause matching rabbit_shovel_parameters:'-lookup_indices/2-lc$^0/1-0-'(#{}) line 430 in context start_error

2018-11-26 08:16:47.712 [error] <0.5040.0> CRASH REPORT Process <0.5040.0> with 0 neighbours crashed with reason: no case clause matching {error,{{shutdown,{failed_to_start_child,{<<"/">>,<<"shovel1">>},{function_clause,
[{rabbit_shovel_parameters,'-lookup_indices/2-lc$^0/1-0-',[#{}],[{file,"src/rabbit_shovel_parameters.erl"},{line,430}]},
{rabbit_shovel_parameters,parse_amqp091_dest,4,
[{file,"src/rabbit_shovel_parameters.erl"},{line,316}]},
{rabbit_shovel_parameters,parse,3,[{file,"src/rabbit_shovel_parameters.erl"},{line,241}]},
{rabbit_shovel_worker,init,1,[{file,"src/rabbit_shovel_worker.erl"},{line,53}]},{gen_server2,init_it,...},...]}}},...}} in rabbit_shovel_dyn_worker_sup_sup:start_child/2 line 45
```

**Comment by michaelklishin** [comment]
Also relevant: 4afc8ac1be429ed875285a196c020c560adac4b2.

**Comment by michaelklishin** [comment]
This has been addressed by 2d73d979e20bd1b8c25aaf08ce01f733abf653e8 months ago but only now backported to `v3.7.x.`. It will ship in `3.7.10`. #47 adds additional tests in this area.
